src/main/scala/uk/gov/hmrc/SbtSassCompiler.scala

```diff
@@ -63,49 +63,69 @@ object SbtSassCompiler extends AutoPlugin {
+                    .resolve(sourcePath.relativize(sassFile.toPath))
+                    .resolveSibling(sassFile.base + ".css")
+                  Right((compiledCss, cssFile))
+                case Failure(compilationError) => Left(compilationError)
+              }
             }
-          }
 
-          val errors: Seq[Throwable]           = eitherCompiledCssFiles.flatMap(_.left.toOption)
-          val compiledCss: Seq[(String, Path)] = eitherCompiledCssFiles.flatMap(_.right.toOption)
-
-          if (errors.nonEmpty) {
-            throw new Exception(
-              s"${errors.length} error(s) whilst compiling Sass files\n${errors.mkString("\n")}"
-            )
-          } else {
-            val cssFiles: Seq[File] = compiledCss map { case (css, cssFile) =>
-              IO.createDirectory(cssFile.getParent.toFile)
-              IO.write(cssFile.toFile, css)
-              cssFile.toFile
+            val errors: Seq[Throwable] = eitherCompiledCssFiles.flatMap(_.left.toOption)
+            val compiledCss: Seq[(String, Path)] = eitherCompiledCssFiles.flatMap(_.right.toOption)
+
+            if (errors.nonEmpty) {
+              throw new Exception(
+                s"sbt-sass-compiler: ${errors.length} error(s) whilst compiling Sass files\n${errors.mkString("\n")}"
+              )
+            } else {
+              val cssFiles: Seq[File] = compiledCss map { case (css, cssFile) =>
+                IO.createDirectory(cssFile.getParent.toFile)
+                IO.write(cssFile.toFile, css)
+                cssFile.toFile
+              }
+              val endInstant = System.currentTimeMillis
+
+              logger.info(
+                s"sbt-sass-compiler: Sass compilation done in ${endInstant - startInstant} ms. Number of CSS files generated: ${cssFiles.length}"
+              )
+              cssFiles
             }
-            val endInstant          = System.currentTimeMillis
+          }.get
+        }
 
-            logger.info(
-              s"Sass compilation done in ${endInstant - startInstant} ms. Number of CSS files generated: ${cssFiles.length}"
-            )
-            cssFiles
+        // this will recompile all Sass anytime any Sass file has changed since last run of task
+        import sbt.util.CacheImplicits._
+        val cachedCompiledSassFiles: Seq[sbt.ModifiedFileInfo] => Seq[sbt.File] =
+          Tracked.inputChanged[Seq[ModifiedFileInfo], Seq[File]](streamsValue.cacheStoreFactory.make("input")) {
+            (anySassFileChanged: Boolean, _) =>
+              if (anySassFileChanged) {
+                logger.info("sbt-sass-compiler: Uncompiled Sass files found")
+                compileSassToCssFiles()
+              } else {
+                logger.info("sbt-sass-compiler: No changes to Sass files, sbt-sass-compiler skipping compilation")
+                Nil
+              }
           }
-        }.get
+        cachedCompiledSassFiles(sassFilesFound.map(FileInfo.lastModified(_)))
```

Comment:
I think one more test we need is to check if sass is recompiled if a partial is modified - I think here because we're using sassFilesFound which excludes partials (sass files with names starting with "_") we might not recompile sass if just a partial is changed

Reply:
sbt-sassify has a test for this https://github.com/irundaia/sbt-sassify/blob/master/src/sbt-test/incremental/changed-imported-file/test though it seems to do what we do https://github.com/irundaia/sbt-sassify/blob/master/src/main/scala/org/irundaia/sbt/sass/SbtSassify.scala#L76 so I might be mistaking how stuff is working somehow